sql/item_float_interchange.h

```diff
@@ -0,0 +1,130 @@
+#ifndef ITEM_FLOAT_INTERCHANGE_INCLUDED
+#define ITEM_FLOAT_INTERCHANGE_INCLUDED
+
+/* Copyright (c) 2018 MariaDB Foundation
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301  USA */
+
+#include "item.h"
+#include "my_float.h"
+
+
+#if float32_exists
+/*************************************************************************
+  Item_func_interchange_float32_to_uint32
+*************************************************************************/
+
+class Item_func_interchange_float32_to_uint32 : public Item_long_func {
+  bool check_arguments() const {
+    return check_argument_types_can_return_real(0, arg_count);
+  }
+
+ public:
+  Item_func_interchange_float32_to_uint32(THD *thd, Item *a)
+      : Item_long_func(thd, a) {}
+  ulonglong val_uint();
+  longlong val_int();
+  const char *func_name() const { return "interchange_float32_to_uint32"; }
+  bool fix_length_and_dec() {
+    decimals= 0;
+    max_length= MAX_INT_WIDTH;
+    maybe_null= args[0]->maybe_null;
+    unsigned_flag= 1;
+    return FALSE;
+  }
+  Item *get_copy(THD *thd) {
+    return get_item_copy<Item_func_interchange_float32_to_uint32>(thd, this);
+  }
+};
+
+/*************************************************************************
+  Item_func_interchange_uint32_to_float32
+*************************************************************************/
+
+class Item_func_interchange_uint32_to_float32 : public Item_real_func {
+  bool check_arguments() const {
+    return check_argument_types_can_return_int(0, arg_count);
+  }
+
+ public:
+  Item_func_interchange_uint32_to_float32(THD *thd, Item *a)
+      : Item_real_func(thd, a) {}
+  double val_real();
+  const char *func_name() const { return "interchange_uint32_to_float32"; }
+  bool fix_length_and_dec() {
+    maybe_null= args[0]->maybe_null;
```

Comment:
Sorry for confusing you, I overlooked one detail yesterday.
These two methods:
- Item_func_interchange_uint32_to_float32::val_real()
- Item_func_interchange_uint64_to_float64::val_real()
can set null_value to true when the integer argument is not null, but the result is not isfinite().
So,
- Item_func_interchange_uint32_to_float32::fix_length_and_dec()
- Item_func_interchange_uint64_to_float64::fix_length_and_dec()
should actually always set maybe_null to true, not matter what args[0]->maybe_null is.

Note, the other two classes (converting from real to integer) never return NULL for not-NULL input.
So these methods:
- Item_func_interchange_float32_to_uint32::fix_length_and_dec()
- Item_func_interchange_float64_to_uint64::fix_length_and_dec()
should do:
  maybe_null= args[0]->maybe_null;